File: examples/Scala/rtrouter.scala
/*
 *  Cross-connected ROUTER sockets addressing each other
 *
 *
 *  @Author:     Giovanni Ruggiero
 *  @Email:      giovanni.ruggiero@gmail.com
*/

import org.zeromq.ZMQ
import ZHelpers._

object rtrouter  {
  def main(args : Array[String]) {
		val ctx = ZMQ.context(1)
		val worker = ctx.socket(ZMQ.ROUTER)
    worker.setIdentity("WORKER" getBytes)
		worker.bind("tcp://*:5555")

		val server = ctx.socket(ZMQ.ROUTER)
    server.setIdentity("SERVER" getBytes)
		server.connect("tcp://localhost:5555")

    //  Wait for the worker to connect so that when we send a message
    //  with routing envelope, it will actually match the worker...
		Thread.sleep(1000)

		server.send("WORKER" getBytes, ZMQ.SNDMORE)
		server.send("" getBytes, ZMQ.SNDMORE)
		server.send("send to worker" getBytes,0)
		dump(worker)

		worker.send("SERVER" getBytes, ZMQ.SNDMORE)
		worker.send("" getBytes, ZMQ.SNDMORE)
		worker.send("send to server" getBytes,0)
		dump(server)

	}
}

File: examples/Lua/rtrouter.lua
--
--  Cross-connected ROUTER sockets addressing each other
--
--  Author: Robert G. Jakabosky <bobby@sharedrealm.com>
--
require"zmq"
require"zhelpers"

local context = zmq.init(1)

local worker = context:socket(zmq.ROUTER)
worker:setopt(zmq.IDENTITY, "WORKER")
worker:bind("ipc://rtrouter.ipc")

local server = context:socket(zmq.ROUTER)
server:setopt(zmq.IDENTITY, "SERVER")
server:connect("ipc://rtrouter.ipc")

--  Wait for the worker to connect so that when we send a message
--  with routing envelope, it will actually match the worker...
s_sleep (1000)

server:send("WORKER", zmq.SNDMORE)
server:send("", zmq.SNDMORE)
server:send("send to worker")
s_dump     (worker)

worker:send("SERVER", zmq.SNDMORE)
worker:send("", zmq.SNDMORE)
worker:send("send to server")
s_dump     (server)

worker:close()
server:close()
context:term()


